### PR TITLE
Scorer overloads

### LINKED ISSUE #770: Add overloads to scorers so they also work on relevant, pdb, smiles input etc.
Currently a lot of the scorers only take a `DockingResults` object. Instead we can decouple or add an overload for the equivalent raw input.

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #770 

## Developers certificate of origin
- [X] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by hmacdope** [comment]
@jthorton could I grab a review here also?

File: asapdiscovery-docking/asapdiscovery/docking/scorer.py [hunk 1]
```diff
@@ -394,21 +663,24 @@ def get_ml_scorer_cls_from_model_type(model_type: ModelType):
...
-    Score from a combination of other scorers
+    Score from a combination of other scorers, the scorers must share an input type,
     """
 
     scorers: list[ScorerBase] = Field(..., description="Scorers to score with")
 
     def score(
         self,
-        inputs: list[DockingResult],
+        inputs: list[Any],
```
Comment (jthorton) [inline asapdiscovery-docking/asapdiscovery/docking/scorer.py:673]: Should this be a union of the supported types list[DockingResult | Complex | Path] ?

File: asapdiscovery-docking/asapdiscovery/docking/tests/test_scorers.py [hunk 2]
```diff
@@ -9,49 +9,52 @@
...
+# parametrize over fixtures
+@pytest.mark.parametrize(
+    "data_fixture", ["results_simple_nolist", "complex_simple", "pdb_simple"]
+)
+@pytest.mark.parametrize("return_df", [True, False])
 @pytest.mark.parametrize("use_dask", [True, False])
-def test_chemgauss_scorer(results_multi, use_dask):
+def test_chemgauss_scorer(use_dask, return_df, data_fixture, request):
+    data = request.getfixturevalue(data_fixture)
+    print(type(data))
```
Comment (jthorton) [inline asapdiscovery-docking/asapdiscovery/docking/tests/test_scorers.py:20]: Comment out print

File: asapdiscovery-docking/asapdiscovery/docking/scorer.py [hunk 3]
```diff
@@ -136,14 +204,36 @@ def _score() -> list[DockingResult]: ...
 
     def score(
         self,
-        inputs: Union[list[DockingResult], list[Path]],
+        inputs: list[Any],
```
Comment (jthorton) [inline asapdiscovery-docking/asapdiscovery/docking/scorer.py:207]: add the supported inputs to the type hint

**Review by jthorton — APPROVED** [review]
Great work, starting to really like overloads!

File: asapdiscovery-docking/asapdiscovery/docking/scorer.py (hunk 1 above)
Comment (hmacdope) [reply asapdiscovery-docking/asapdiscovery/docking/scorer.py:673]: Yes good catch, I was lazy.

**Pull request merged**